cluster/addons/dns/skydns-rc.yaml.in

```diff
@@ -20,11 +20,12 @@ desiredState:
         dnsPolicy: "Default"  # Don't use cluster DNS.
         containers:
           - name: etcd
-            image: quay.io/coreos/etcd:latest
+            image: quay.io/coreos/etcd:v2.0.3
             command: [
-                    "/etcd",
-                    "-bind-addr=127.0.0.1",
-                    "-peer-bind-addr=127.0.0.1",
+                    # entrypoint = "/etcd",
```

Comment:
Since I'm not that knowledgeable around Docker I can't answer with certainty, but here is how I understand it:
- both etcd images (latest and 2.0.3) have ENTRYPOINT defined as /etcd and empty cmd (https://quay.io/repository/coreos/etcd?image=201a4476d171)
- such configuration causes docker run etcd_image to just execute ENTRYPOINT and cmd (https://docs.docker.com/reference/builder/#entrypoint)
- I'm not sure why, but running $ etcd /etcd will start etcd without any error; moreover $ etcd /etcd -listen-client-urls=http://0.0.0.0:2379,http://0.0.0.0:4001 will not really run etcd with the command line flag as expect
- I've tested manually configuration in my PR and it works
- I believe that previous configuration wasn't really correct; I think it only worked just because the provided flags were equal to the defaults (otherwise the provided flags would not have any effect).

Regarding your request to stick with "bin/sh". I don't think it will really work. As I understand one of the reasons for ENTRYPOINT equal to /etcd is for users to use that entrypoint and just configure the application from command line. Entrypoint can be override by --entrypoint docker flag, however it won't work in case of etcd image since it doesn't contain /bin/sh.